Tests/RugbyTests/Core/Dependencies/VersionParserTests.swift

```diff
@@ -0,0 +1,94 @@
+        XCTAssertEqual(ghVersion.patch, parsedVersion.patch)
+        XCTAssertEqual(ghVersion.prerelease, parsedVersion.prerelease)
+        XCTAssertEqual(ghVersion.build, parsedVersion.build)
+    }
+    
+    func testParseValidBetaVersion() throws {
+        let version = "1.0.0b"
+        let ghVersion = GitHubUpdaterVersion(major: 1, minor: 0, patch: 0, prerelease: GitHubUpdaterVersion.Prerelease.beta, build: 0)
+        let parsedVersion = try self.parser.parse(version)
+        XCTAssertEqual(ghVersion.major, parsedVersion.major)
+        XCTAssertEqual(ghVersion.minor, parsedVersion.minor)
+        XCTAssertEqual(ghVersion.patch, parsedVersion.patch)
+        XCTAssertEqual(ghVersion.prerelease, parsedVersion.prerelease)
+        XCTAssertEqual(ghVersion.build, parsedVersion.build)
+    }
+    
+    func testParseValidBuildVersion() throws {
+        let version = "1.0.0b1"
+        let ghVersion = GitHubUpdaterVersion(major: 1, minor: 0, patch: 0, prerelease: GitHubUpdaterVersion.Prerelease.beta, build: 1)
+        let parsedVersion = try self.parser.parse(version)
+        XCTAssertEqual(ghVersion.major, parsedVersion.major)
+        XCTAssertEqual(ghVersion.minor, parsedVersion.minor)
+        XCTAssertEqual(ghVersion.patch, parsedVersion.patch)
+        XCTAssertEqual(ghVersion.prerelease, parsedVersion.prerelease)
+        XCTAssertEqual(ghVersion.build, parsedVersion.build)
+    }
+    
+    func testParseInvalidBuildVersion() throws {
+        let version = "1.0.0ba"
+        let ghVersion = GitHubUpdaterVersion(major: 1, minor: 0, patch: 0, prerelease: GitHubUpdaterVersion.Prerelease.beta, build: 1)
+        let parsedVersion = try self.parser.parse(version)
+        XCTAssertEqual(ghVersion.major, parsedVersion.major)
+        XCTAssertEqual(ghVersion.minor, parsedVersion.minor)
+        XCTAssertEqual(ghVersion.patch, parsedVersion.patch)
+        XCTAssertEqual(ghVersion.prerelease, parsedVersion.prerelease)
+        XCTAssertNotEqual(ghVersion.build, parsedVersion.build)
+    }
+    
+    func testParseInvalidBetaVersion() throws {
+        let version = "1.0.0a1"
+        let ghVersion = GitHubUpdaterVersion(major: 1, minor: 0, patch: 0, prerelease: GitHubUpdaterVersion.Prerelease.beta, build: 1)
+        let parsedVersion = try self.parser.parse(version)
+        XCTAssertEqual(ghVersion.major, parsedVersion.major)
+        XCTAssertEqual(ghVersion.minor, parsedVersion.minor)
+        XCTAssertEqual(ghVersion.patch, parsedVersion.patch)
+        XCTAssertNotEqual(ghVersion.prerelease, parsedVersion.prerelease)
+        XCTAssertNotEqual(ghVersion.build, parsedVersion.build)
+    }
+    
+    func testInvalidShortVersionParse() {
+        let version = "1.0"
+        XCTAssertThrowsError(try self.parser.parse(version)) { error in
+            XCTAssertEqual((error as! VersionParserError).localizedDescription, VersionParserError.incorrectVersion(version).localizedDescription)
```

Comment:
1. It's better to keep expected value in the beginning of test;
2. We can skip `self` here;
3. It's dangerous to use force unwrap `!` in tests. Here we can use `?` without any problems.
   We sure that expected value isn't nil and anyway we will catch nil error in block.
```suggestion
        let version = "1.0"
        let expectedLocalizedDescription = VersionParserError.incorrectVersion(version).localizedDescription
        
        // Act & Assert
        XCTAssertThrowsError(try parser.parse(version)) { error in
            XCTAssertEqual((error as? VersionParserError).localizedDescription, expectedLocalizedDescription)
```